src/group_theory/subgroup.lean

```diff
@@ -1182,6 +1182,12 @@ def cod_restrict (f : G →* N) (S : subgroup N) (h : ∀ x, f x ∈ S) : G →*
   map_one' := subtype.eq f.map_one,
   map_mul' := λ x y, subtype.eq (f.map_mul x y) }
 
+/-- The range of an injective group homomorphism is isomorphic to its domain. -/
+noncomputable def injective_range_equiv {G H : Type*} [group G] [group H] {f : G →* H}
+  (hf : function.injective f) : f.range ≃* G :=
+(mul_equiv.of_bijective (f.cod_restrict f.range (λ x, ⟨x, rfl⟩))
+  ⟨λ x y h, hf (subtype.ext_iff.mp h), by { rintros ⟨x, y, rfl⟩, exact ⟨y, rfl⟩ }⟩).symm
```

Comment:
We have this API already for `linear_equiv` etc; it should be called `mul_equiv.of_injective : G ≃* f.range`. Could you add the `mul_equiv.of_left_inverse` variant too?

Reply:
https://github.com/leanprover-community/mathlib/blob/b401f074d14a59c595b3e9f2a995fd305ff20c85/src/linear_algebra/basic.lean#L2007-L2027